isofit/utils/template_construction.py

```diff
@@ -1686,3 +1719,115 @@ def reassemble_cube(matching_indices: np.array, paths: Pathnames):
             output_mm[matching_indices == _st, ...] = input_ds.open_memmap(
                 interleave="bip"
             )[:, :, : int(header["bands"])].copy()[:, 0, :]
+
+
+def make_surface_config(
+    paths: Pathnames,
+    surface_category="multicomponent_surface",
+    pressure_elevation=None,
+    elevation_lut_grid=[],
+    presolve=False,
+    surface_mapping: dict = None,
+):
+    """
+    Constructs the surface component of the config
+    Args:
+        paths: Pathnames object with all key values passed from apply_oe
+        surface_category: Base surface category
+    Returns:
+        surface_config_dict: Dictionary with all surface parameters and file
+                             locations.
+    """
+
+    # Initialize config dict
+    surface_config_dict = {
+        "multi_surface_flag": False,
+    }
+
+    # Check to see if a classification file is being propogated
+    if paths.surface_class_file and not presolve:
+        surface_config_dict["Surfaces"] = {}
+        surface_config_dict["surface_class_file"] = paths.surface_class_file
+
+        if vars(paths).get("subs_class_path", None):
+            surface_config_dict["sub_surface_class_file"] = paths.subs_class_path
+
+        surface_config_dict["multi_surface_flag"] = True
+
+        surface_class_ds = envi.open(envi_header(paths.surface_class_file))
+
+        # Get the class mapping. Tried to build in some insensitivity here.
+        # Will use first option it hits
+        options = [
+            "mapping",
+            "Mapping",
+            "class names",
+            "Class names",
+            "class",
+            "Class",
+            "surfaces",
+            "Surfaces",
+            "surface names" "Surface names",
+        ]
+        class_mapping = None
+        for option in options:
+            if class_mapping:
+                continue
+            class_mapping = surface_class_ds.metadata.get(option)
+
+        # mapping name to surface name - terrible way to do this
+        # Could house this in a standalone file and call it in
+        if not surface_mapping:
+            surface_mapping = {
+                "water": "glint_model_surface",
+                "glint": "glint_model_surface",
+                "land": "multicomponent_surface",
+                "nonwater": "multicomponent_surface",
+                "cloud": "multicomponent_surface",
+                "uniform_surface": "multicomponent_surface",
+            }
```

Comment:
@evan-greenbrg What are `nonwater` and `uniform_surface`? Aren't they covered by the `land `key?

Reply:
If not provided, the surface_mapping is hard-coded so that the program can read the classification name from the file and match it with the Isofit surface name. This means that the user doesn't have to use the Isofit surface names when constructing the classification image. 

Because there is some less than ideal specificity that is required, I've used keys here that include multiple potential class names for a given surface model (e.g. water and glint as well as land and nonwater). All keys in the surface_mapping object aren't used in the processing, just the ones present in the image.

I have an associated Jupyter Notebook and data cube that helps explain the set up for the classification file. I need to work on merge in into Isotuts and get the data cube into a place where it can be downloaded.

Reply:
That's a great idea! Having a tutorial notebook for this topic would be really helpful.